hal/src/rtc/rtic/mod.rs

```diff
@@ -0,0 +1,335 @@
+
+    #[hal_cfg(any("rtc-d11", "rtc-d21"))]
+    crate::__internal_half_period_counting_backend!(
+        RtcBackend, RtcMode1, 1, Compare0, Compare1, Overflow
+    );
+
+    // For SAMx5x mode 0 is the best option
+    #[hal_cfg("rtc-d5x")]
+    use crate::rtc::modes::mode0::{Compare0, Compare1, Overflow};
+
+    #[hal_cfg("rtc-d5x")]
+    crate::__internal_half_period_counting_backend!(
+        RtcBackend, RtcMode0, 0, Compare0, Compare1, Overflow
+    );
+}
+
+pub use backend::RtcBackend;
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __internal_create_rtc_interrupt {
+    ($backend:ident) => {
+        #[no_mangle]
+        #[allow(non_snake_case)]
+        unsafe extern "C" fn RTC() {
+            $crate::rtc::rtic::$backend::interrupt_handler();
+        }
+    };
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __internal_create_rtc_struct {
+    ($name:ident, $backend:ident, $clock_rate:ty) => {
+        /// A `Monotonic` based on the RTC peripheral.
+        pub struct $name;
+
+        impl $name {
+            /// This method must be called only once.
+            pub fn start(rtc: $crate::pac::Rtc) {
+                use $crate::rtc::rtic::rtc_clock::*;
+                $crate::__internal_create_rtc_interrupt!($backend);
+
+                $crate::rtc::rtic::$backend::_start(rtc);
+            }
+        }
+
+        use $crate::rtc::rtic::rtc_clock::RtcClockRate;
+
+        impl $crate::rtic_time::monotonic::TimerQueueBasedMonotonic for $name {
+            type Backend = $crate::rtc::rtic::$backend;
+            type Instant = $crate::fugit::Instant<
+                <Self::Backend as $crate::rtic_time::timer_queue::TimerQueueBackend>::Ticks,
+                1,
+                { <$clock_rate>::RATE_HZ },
+            >;
+            type Duration = $crate::fugit::Duration<
+                <Self::Backend as $crate::rtic_time::timer_queue::TimerQueueBackend>::Ticks,
+                1,
+                { <$clock_rate>::RATE_HZ },
+            >;
+        }
+
+        $crate::rtic_time::impl_embedded_hal_delay_fugit!($name);
+        $crate::rtic_time::impl_embedded_hal_async_delay_fugit!($name);
+    };
+}
+
+/// Create an RTIC v2 monotonic that uses the RTC.
+///
+/// See the [`rtic`](crate::rtc::rtic) module for details.
+#[macro_export]
+macro_rules! rtc_monotonic {
+    ($name:ident, $clock_rate: ty) => {
+        $crate::__internal_create_rtc_struct!($name, RtcBackend, $clock_rate);
+    };
+}
+
+/// This function was copied from the private function in `rtic-monotonics`,
+/// so that should be used when the monotonics move there.
+const fn cortex_logical2hw(logical: u8, nvic_prio_bits: u8) -> u8 {
+    ((1 << nvic_prio_bits) - logical) << (8 - nvic_prio_bits)
+}
+
+/// This function was copied from the private function in `rtic-monotonics`.
+unsafe fn set_monotonic_prio(prio_bits: u8, interrupt: impl cortex_m::interrupt::InterruptNumber) {
+    extern "C" {
+        static RTIC_ASYNC_MAX_LOGICAL_PRIO: u8;
```

Comment:
An extern static is pretty unusual in Rust. I've personally scratched my head for a while on exactly this `RTIC_ASYNC_MAX_LOGICAL_PRIO` thing not so long ago. I would definitely explain in the docs somewhere what this u8 means, and that `rtic` will initialize the static, but users writing rtic-less apps should set the static themselves using

```rs
#[no_mangle]
pub static RTIC_ASYNC_MAX_LOGICAL_PRIO: u8 = (something);
```